src/main/java/thewaypointers/trafficsimulator/TrafficSimulatorManager.java

```diff
@@ -1,76 +0,0 @@
-package thewaypointers.trafficsimulator;
-
-import thewaypointers.trafficsimulator.common.IStateProvider;
-import thewaypointers.trafficsimulator.common.WorldStateDTO;
-import thewaypointers.trafficsimulator.common.helpers.JunctionTestProvider;
-import thewaypointers.trafficsimulator.gui.MainFrame;
-
-import javax.swing.*;
-
-public class TrafficSimulatorManager extends Thread {
-
-    public static final float VEHICLE_MOVEMENT_SPEED = 2;
-    public static final float STATES_PER_SECOND = 1;
-    private static MainFrame mainFrame;
-    private static IStateProvider simulation;
-    private WorldStateDTO worldState;
-    private boolean stop = true;
-    private boolean isSleep = true;
-
-
-
-    private JButton start_pauseButton;
```

Comment:
Why did it have the reference to a button? This is controller code, it doesn't need to have a button existing to work. Changing the label can be done in the GUI logic, and if you want it always synchronized with the actual state of the controller, create some way to get this information from the controller, but don't make the controller manage buttons in the GUI.